playwright_tests/helpers/fixtures.ts

```diff
@@ -48,8 +49,23 @@ export const test = base.extend<{
       await page.getByLabel('No').check();
       await page.getByRole('button', { name: 'Continue' }).click();
       await page.locator('#confirm-terms-and-conditions').click();
+      const spinner = page.locator('div.spinner-wrapper');
       await page.getByRole('button', { name: 'Confirm and submit' }).click();
-      await expect(page.getByRole('heading', { name: 'Registration details submitted' })).toBeVisible();
+      await page.waitForTimeout(2000);
+      await spinner.waitFor({ state: 'hidden', timeout: 30_000 });
+      for (let i = 0; i < 6; i++) {
```

Comment:
6? might be better to have a small delay e.g 50-100ms in the loop, and fewer retries. Also 30 sec delay is quite long

Reply:
What will happen usually is it will work quickly as posssible but everything should be in place to ensure that there are no problems if there are loading/performance issues